### PR TITLE
Refactor/draft folders

### PR DESCRIPTION
**Pull request by csc-jm**
### Description

As discussed, a refactoring to the way drafts are stored in folders was in place. The current idea with the `publish_folder` endpoint in this PR is that the drafts in the `drafts` key of the folder that is to be published are removed from the database and the folder is simultaneously patched so that the drafts are removed from the folder.

### Type of change

- [x] Breaking change (or not exactly breaking but somewhat altering and requires a change to frontend)

### Changes Made

- added `drafts` key to the folders
- updated to the OpenApi specs and json schema
- created new endpoint for publishing a folder
- updated the existing integration test for folders
- other general style fixes 

### Testing

- [x] Unit Tests
- [x] Integration Tests

### CODE REVIEW

File: metadata_backend/api/handlers.py [hunk 1]
```diff
@@ -366,6 +366,37 @@ async def patch_folder(self, req: Request) -> Response:
         LOG.info(f"PATCH folder with ID {folder} was successful.")
         return web.Response(body=body, status=200, content_type="application/json")
 
+    async def publish_folder(self, req: Request) -> Response:
+        """Update object folder specifically into published state.
+
+        :param req: PATCH request
```
Comment (blankdots) [inline metadata_backend/api/handlers.py:372]: this is a `patch` but in routes it looks like a `.get` .... maybe we can settle on something

File: metadata_backend/server.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ async def init() -> web.Application:
         web.get("/folders/{folderId}", rest_handler.get_folder),
         web.patch("/folders/{folderId}", rest_handler.patch_folder),
         web.delete("/folders/{folderId}", rest_handler.delete_folder),
+        web.get("/folders/publish/{folderId}", rest_handler.publish_folder),
```
Comment (blankdots) [inline metadata_backend/server.py:75]: how about `/publish/{folderId}` ?

Comment (blankdots) [reply metadata_backend/server.py:75]: and a `patch`

Comment (blankdots) [reply metadata_backend/server.py:75]: also this might create an edge case with where `/folders/{folderId}` == /folders/publish`

**Review by blankdots — CHANGES_REQUESTED** [review]
some things, will test later

**Review by blankdots — APPROVED** [review]
